subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -20,13 +20,30 @@
 
 LOG_MODULE_DECLARE(bt_ots, CONFIG_BT_OTS_LOG_LEVEL);
 
+/* In order to transmit the maximum size we choose CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU
+ * as the buffer size. However, directory listing encoding can only happen at
+ * directory entry boundary so we must ensure the MTU value is rounded up to the
+ * nearest directory entry size to not overflow our buffer. Also since we can only
+ * encode at object entry boundary, we need to make sure we have at least room for
+ * two entries so entries can be read from the middle.
+ */
+#define OTS_DIR_LIST_BUFFER_SIZE MAX(2 * DIR_LIST_OBJ_RECORD_MAX_SIZE, \
+	DIR_LIST_OBJ_RECORD_MAX_SIZE * \
+	ceiling_fraction(CONFIG_BT_OTS_L2CAP_CHAN_TX_MTU, DIR_LIST_OBJ_RECORD_MAX_SIZE))
+
+struct bt_ots_dir_list {
+	struct net_buf_simple net_buf;
+	struct bt_gatt_ots_object *dir_list_obj;
+	uint8_t _content[OTS_DIR_LIST_BUFFER_SIZE];
+};
```

Comment:
This could/should be located in `ots_dir_list_internal.h`

Reply:
I had intentionally moved it here as `struct bt_ots_dir_list` is now opaque after introduction of `bt_ots_dir_list_is_idle`.

Reply:
Right, that made sense when we had 2 different implementations of `bt_ots_dir_list`, but that is no longer the case :)

Reply:
That is fair, I will move it back :).